Plugins/ExternalTexture/Source/ExternalTexture_Metal.mm

```diff
@@ -0,0 +1,200 @@
+        { MTLPixelFormatRGBA8Snorm,                     MTLPixelFormatInvalid                       }, // RGBA8S
+        { MTLPixelFormatRGBA16Unorm,                    MTLPixelFormatInvalid                       }, // RGBA16
+        { MTLPixelFormatRGBA16Sint,                     MTLPixelFormatInvalid                       }, // RGBA16I
+        { MTLPixelFormatRGBA16Uint,                     MTLPixelFormatInvalid                       }, // RGBA16U
+        { MTLPixelFormatRGBA16Float,                    MTLPixelFormatInvalid                       }, // RGBA16F
+        { MTLPixelFormatRGBA16Snorm,                    MTLPixelFormatInvalid                       }, // RGBA16S
+        { MTLPixelFormatRGBA32Sint,                     MTLPixelFormatInvalid                       }, // RGBA32I
+        { MTLPixelFormatRGBA32Uint,                     MTLPixelFormatInvalid                       }, // RGBA32U
+        { MTLPixelFormatRGBA32Float,                    MTLPixelFormatInvalid                       }, // RGBA32F
+        { MTLPixelFormat(40/*B5G6R5Unorm*/),            MTLPixelFormatInvalid                       }, // R5G6B5
+        { MTLPixelFormat(42/*ABGR4Unorm*/),             MTLPixelFormatInvalid                       }, // RGBA4
+        { MTLPixelFormat(41/*A1BGR5Unorm*/),            MTLPixelFormatInvalid                       }, // RGB5A1
+        { MTLPixelFormatRGB10A2Unorm,                   MTLPixelFormatInvalid                       }, // RGB10A2
+        { MTLPixelFormatRG11B10Float,                   MTLPixelFormatInvalid                       }, // RG11B10F
+        { MTLPixelFormatInvalid,                        MTLPixelFormatInvalid                       }, // UnknownDepth
+        { MTLPixelFormatDepth32Float,                   MTLPixelFormatInvalid                       }, // D16
+        { MTLPixelFormatDepth32Float,                   MTLPixelFormatInvalid                       }, // D24
+        { MTLPixelFormat(255/*Depth24Unorm_Stencil8*/), MTLPixelFormatInvalid                       }, // D24S8
+        { MTLPixelFormatDepth32Float,                   MTLPixelFormatInvalid                       }, // D32
+        { MTLPixelFormatDepth32Float,                   MTLPixelFormatInvalid                       }, // D16F
+        { MTLPixelFormatDepth32Float,                   MTLPixelFormatInvalid                       }, // D24F
+        { MTLPixelFormatDepth32Float,                   MTLPixelFormatInvalid                       }, // D32F
+        { MTLPixelFormatStencil8,                       MTLPixelFormatInvalid                       }, // D0S8
+    };
+    BX_STATIC_ASSERT(bgfx::TextureFormat::Count == BX_COUNTOF(s_textureFormat));
+}
+
+// clang-format on
+
+namespace Babylon::Plugins
+{
+    class ExternalTexture::Impl
+    {
+    public:
+        Impl(Graphics::TextureT ptr)
+        {
+            m_ptr = ptr;
+            [m_ptr retain];
```

Comment:
We've had so many memory management issues on iOS, can we just enable ARC for the ExternalTexture lib? We do this in other BN libs in CMakeLists.txt with:
```
if (APPLE)
    set_target_properties(NativeTracing PROPERTIES
        XCODE_ATTRIBUTE_CLANG_ENABLE_OBJC_ARC YES
    )
endif()
```

Reply:
I think so. That mean I don't need to call retain/release anymore, right?

Reply:
Yes that should be correct. You can verify by leaving it there, in which case you should get a compile error when ARC is enabled.